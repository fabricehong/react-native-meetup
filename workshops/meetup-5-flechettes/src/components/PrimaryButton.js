import React, { Component } from 'react';
import { Text, View, Alert, StyleSheet, TouchableHighlight } from 'react-native';


const PrimaryButton = (props)=>{
    const {label, onPress} = props;
    return (
        <TouchableHighlight style={styles.button} onPress={onPress}>
        <Text style={styles.text} >{label}</Text>
        </TouchableHighlight>
    )
}

const styles = StyleSheet.create({
    button: {
      backgroundColor:'#aaaaff',
      margin: 10,
      padding: 10,
      
    },
    text: {
      textAlign: 'center'
    }
  });
  
export default PrimaryButton;


/* 
export default class App extends Component {
  render() {
    return (
      <View>
        {
          Array.from(Array(n).keys()).map(number => <ScoreButton/>)
        }
        <Text>Hello</Text>
      </View>
    )
  }
}
*/