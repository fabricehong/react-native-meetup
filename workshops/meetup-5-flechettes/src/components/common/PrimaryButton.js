import React from 'react';
import {StyleSheet, Text, TouchableHighlight} from 'react-native';

const PrimaryButton = (props) => {
  const {label, onPress} = props;
  return (
    <TouchableHighlight style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{label}</Text>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#aaaaff', margin: 10, padding: 10,

  }, text: {
    textAlign: 'center',
  },
});

export default PrimaryButton;