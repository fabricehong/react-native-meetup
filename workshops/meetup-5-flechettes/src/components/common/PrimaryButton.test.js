import React from 'react';
import PrimaryButton from './PrimaryButton';
import renderer from 'react-test-renderer';

describe('PrimaryButton', () => {
  test('should renders correctly', () => {
    const tree = renderer.create(<PrimaryButton/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});