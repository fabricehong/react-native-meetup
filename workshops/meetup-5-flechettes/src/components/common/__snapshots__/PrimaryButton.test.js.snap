// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton should renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#aaaaff",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
  />
</View>
`;
